File: dataset/human/c/file_3225.c
#include <stdio.h>

int main(void){
  int k,n;
  scanf("%d %d",&k,&n);
  int a[n],b[n];
  for(int i=0;i<n;i++){
    scanf("%d",&a[i]);
  }
  b[0]=k+a[0]-a[n-1];
  int temp = b[0];
  for(int i=1;i<n;i++){
    b[i] = a[i]-a[i-1];
  }
  for(int i =0;i<n;i++){
    if (temp < b[i]){
      temp = b[i];
    }
  }
  printf("%d",k-temp);
  return 0;

}

File: dataset/human/python/file_3225.py
A = int(input())

def factorization(n):
    arr = []
    temp = n
    for i in range(2, int(-(-n**0.5//1))+1):
        if temp%i==0:
            cnt=0
            while temp%i==0:
                cnt+=1
                temp //= i
            arr.append([i, cnt])

    if temp!=1:
        arr.append([temp, 1])

    if arr==[]:
        arr.append([n, 1])

    return arr

B = factorization(A)
N = len(B)
ans = 0
for n in range(N):
  m =1
  b = B[n][1]
  while b >= ((m+1)*(m+2)//2):
    m+=1
  ans += m
if A==1:
  print(0)
else:
  print(ans)
